urllib3/response.py

```diff
@@ -327,9 +384,14 @@ def read(self, amt=None, decode_content=None, cache_content=False):
                     # no harm in redundantly calling close.
                     self._fp.close()
                     flush_decoder = True
+                    if (self.strict_content_length and
+                       self.length_remaining not in (0, None)):
+                        raise IncompleteRead(self._fp_bytes_read, self.length_remaining)
```

Comment:
So here's a note: this only fires for reads with lengths. If you call `read()`, this error will never fire. Should we enforce this for those reads too?